### PR TITLE
Initial commit

### PR DESCRIPTION
**Pull request by jeevan-kamkar**

### CODE REVIEW

**Review by mohamed-shahid-afrid — APPROVED** [review]

File: PROJECT [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+# This file is used to track the info used to scaffold your project
+# and allow the plugins properly work.
+# More info: https://book.kubebuilder.io/reference/project-config.html
+domain: odimra
+layout:
+- go.kubebuilder.io/v3
+plugins:
+  manifests.sdk.operatorframework.io/v2: {}
+  scorecard.sdk.operatorframework.io/v2: {}
+projectName: bmc-operator
```
Comment (amar-shalgar) [inline PROJECT]: Check if this also needs to change

File: config/constants/constants.go [hunk 2]
```diff
@@ -0,0 +1,79 @@
...
+
+package constants
+
+const (
+	// FirmwareSimpleUpdateURI : uri of firmware simple update API
+	FirmwareSimpleUpdateURI = "/redfish/v1/UpdateService/Actions/UpdateService.SimpleUpdate"
+	// FirmwareStartUpdateURI : uri of firmware start update API
+	FirmwareStartUpdateURI = "/redfish/v1/UpdateService/Actions/UpdateService.StartUpdate"
+	// BMCOPERATOR : bmc-operator folder name
+	BMCOPERATOR = "bmc-operator"
```
Comment (amar-shalgar) [inline config/constants/constants.go]: Even this is refering to folder name

**Review by amar-shalgar — CHANGES_REQUESTED** [review]

File: dockerfiles/Dockerfile.bmcoperator [hunk 3]
```diff
@@ -0,0 +1,39 @@
+# Build the manager binary
+FROM golang:1.19.5 as builder
+
+ENV GO111MODULE=auto
+WORKDIR /bmc-operator
```
Comment (amar-shalgar) [inline dockerfiles/Dockerfile.bmcoperator:5]: Check if this needs an update

File: dockerfiles/Dockerfile.bmcoperator [hunk 4]
```diff
@@ -0,0 +1,39 @@
...
+COPY dockerfiles/scripts/ dockerfiles/scripts/
+# Build
+RUN CGO_ENABLED=0 GOOS=linux GOARCH=amd64 go build -a -o manager main/main.go
+
+# Use distroless as minimal base image to package the manager binary
+# Refer to https://github.com/GoogleContainerTools/distroless for more details
+FROM ubuntu:22.04
+
+WORKDIR /
+COPY --from=builder /bmc-operator/manager .
```
Comment (amar-shalgar) [inline dockerfiles/Dockerfile.bmcoperator:28]: Please check if this needs an update

File: dockerfiles/Dockerfile.bmcoperator [hunk 5]
```diff
@@ -0,0 +1,39 @@
...
+COPY --from=builder /bmc-operator/config/keys /bmc-operator/config/keys
+COPY --from=builder /bmc-operator/dockerfiles/scripts/start_bmc_operator.sh .
+RUN mkdir /var/log/operator_logs
+RUN chown -R 2021:2021 /var/log/operator_logs
+RUN groupadd -r -g 2021 odimra && /usr/sbin/useradd -s /bin/bash -u 2021 -m -d /home/odimra -r -g odimra odimra 
+RUN mkdir /etc/config && chown 2021:2021 /etc/config
+RUN chown -R 2021:2021 /bmc-operator/config/keys
```
Comment (amar-shalgar) [inline dockerfiles/Dockerfile.bmcoperator:35]: Check for all occurrences of bmc-operator

**Review by amar-shalgar — CHANGES_REQUESTED** [review]

**Review by amar-shalgar — APPROVED** [review]

**Pull request merged**